### DISCUSSION TITLE
Chapter Creation is Inaccurate

### DISCUSSION THREAD

**Discussion by gattaca44** [yermak/AudioBookConverter#237] [General]
Hi. The program is generally very good, and I like its ability to make M4B audiobooks independent of iTunes or any other software. But I noticed that there is a bug. 

When I create an M4B book from a list of MP3 files that are cut into chapters, the resulting M4B made by AudioBookConverter has inaccurate chapter start times. For instance, when the program made a 54-chapter book for me today, it played fine in general. But when I tried to jump to, say, chapter 42, the audio was a half-second or so ahead of the actual start of the chapter. The higher the number of the chapter, the more the audio was ahead of the chapter. It sounds like AudioBookConverter is adding a fraction of a second to each chapter, so the problem keeps compounding.

I have recreated this problem consistently with several different books. I double-checked the source MP3 files in each case, and they are all perfect. Also, Chapter and Verse makes the books with correct chapter times. So there is some issue with how AudioBookCutter is applying the chapter times. If this can be corrected, it would be great. The chapter discrepancies are the only flaw I have found so far with your program.

**Comment by morpheus2n2** [comment]
HI, sorry I am unable to help however I would like to ask how you made a file with multiple mp3 as when I try I get a single file but all chapter information is lost?

**Comment by yermak** [comment]
Hi,
This issue is discussed many times #21 

Unfortunatly, i can see simple resolution to it, its impact depends a lot on original source file and codec which was used, so it's very diffucult to test all variaty.

As impact is minimal - I don't think it will be fixed soon.

**Reply by gattaca44** [reply]
Sorry to hear you don't consider the issue a priority. Sandreas, the programmer behind M4B-tool, considers it enough a priority that he fixed the issue so that chapter duration is detected perfectly in his program. I tested M4B-tool and it makes perfect chapter lengths every time. I'll use that until Audiobook Converter gets the problem fixed.

**Reply by yermak** [reply]
I found source files where i could repeatedly reproduce this issue. I will have a look shortly if there is anything could be done.

**Reply by yermak** [reply]
@gattaca44, as I mentioned there was a fix which i did for issue #21, but i'm sorry to say this fix was unrelyable and may or may not work depending on pc and hard drive speed. But, I had strong opinion that fix is in place and everything is fine, despite your comment. It took me few hours to understand what is the problem as it was diffucult to reproduce the issue, once i found it, i realised that workaround I put in place 3 years back is no longer needed and there is much simple way to fix it, which should completely illumante the problem for all versions after 5.5.18.

New version should be out in couple of weeks, as I would like to fix few other reported issues.

Thank you very much for your bug report!!!

**Reply by gattaca44** [reply]
Thanks for the reply. I'll look forward to trying out the new version in a couple of weeks. Keep up the good work.
